opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp

```diff
@@ -388,19 +389,111 @@ namespace Opm {
         for (size_t recordNr = 0; recordNr < keyword->size(); recordNr++) {
             DeckRecordConstPtr record = keyword->getRecord(recordNr);
-            const std::string& wellName = record->getItem("WELL")->getTrimmedString(0);
-            WellPtr well = getWell(wellName);
 
+            bool haveCompletionData = false;
             for (size_t i=2; i<7; i++) {
-                if (record->getItem(i)->getInt(0) > 0 ) {
-                    throw std::logic_error("Error processing WELOPEN keyword, specifying specific connections is not supported yet.");
+                if (record->getItem(i)->hasValue(0)) {
+                    haveCompletionData = true;
+                    break;
                 }
             }
+
+
+            const std::string& wellNamePattern = record->getItem("WELL")->getTrimmedString(0);
+            const std::vector<WellPtr>& wells = getWells(wellNamePattern);
+
             WellCommon::StatusEnum status = WellCommon::StatusFromString( record->getItem("STATUS")->getTrimmedString(0));
-            well->setStatus(currentStep, status);
+
+            for (auto wellIter=wells.begin(); wellIter != wells.end(); ++wellIter) {
+                WellPtr well = *wellIter;
+                std::string wellName = well->name();
+                if(haveCompletionData){
+                    CompletionSetConstPtr currentCompletionSet = well->getCompletions(currentStep);
+
+                    CompletionSetPtr newCompletionSet(new CompletionSet( ));
+
+                    Opm::Value<int> I  = getValueItem(record->getItem("I"));
+                    Opm::Value<int> J  = getValueItem(record->getItem("J"));
+                    Opm::Value<int> K  = getValueItem(record->getItem("K"));
+                    Opm::Value<int> C1 = getValueItem(record->getItem("C1"));
+                    Opm::Value<int> C2 = getValueItem(record->getItem("C2"));
+
+                    size_t completionSize = currentCompletionSet->size();
+
+                    for(size_t i = 0; i < completionSize;i++) {
+
+                        CompletionConstPtr completion = currentCompletionSet->get(i);
+
+                        if (C1.hasValue()) {
+                            if (i < (size_t) C1.getValue()) {
+                                newCompletionSet->add(completion);
+                                continue;
+                            }
+                        }
+                        if (C2.hasValue()) {
+                            if (i > (size_t) C2.getValue()) {
+                                newCompletionSet->add(completion);
+                                continue;
+                            }
+                        }
+
+                        int ci = completion->getI();
+                        int cj = completion->getJ();
+                        int ck = completion->getK();
+
+                        if ((I.hasValue() && (!(I.getValue() == ci) || I.getValue() !=0))) {
```

Comment:
With the `getValue<int>()`method we should never assign a zero to the `Value<int>` objects, so it should be sufficient to do:
```C++
 if ((I.hasValue() && (!(I.getValue() == ci)
    newCompletionsSet->add(completion);
...
```